### PR TITLE
Additions to Counters for Yokozuna

### PR DESCRIPTION
**Pull request by lenary**
This is associated with the work in basho/yokozuna#135 - and hence will have to wait until yokozuna supports 1.4.

Firstly, the Yokozuna extractor extracts on a sibling-by-sibling basis, not the entire riak object. Hence it's useful to have a method in `riak_kv_counter` that can give the counter value from a single sibling.

Secondly, we require that Yokozuna merges all CRDTs before doing extraction, so that the most up to date value of the counter is merged. `riak_kv_counter:merge/1` is "safe" to call on a riak_object without counter siblings (ie it won't affect anything), but is quite a heavyweight thing to do. We include a simple check to see if any siblings could be a counter, before merging. This probably needs (micro-)benchmarking to see if it really does increase performance.

### CODE REVIEW

File: src/riak_kv_counter.erl [hunk 1]
```diff
@@ -80,6 +97,17 @@ value(RObj) ->
             riak_kv_pncounter:value(Counter)
     end.
 
+%% @doc Turn a single sibling into a counter value. Used by the YZ Extractor.
+-spec sibling_value(binary()) -> {ok, integer()} | {error, term()}.
+sibling_value(Sibling) ->
+    try
+        PNCounter = riak_kv_counter:from_binary(Sibling),
+        Value = riak_pn_counter:value(PNCounter),
```
Comment (coderoshi) [inline src/riak_kv_counter.erl]: It looks like this is called `riak_kv_pncounter` rather than `riak_pn_counter`

Comment (lenary) [inline src/riak_kv_counter.erl]: Yes, Yes it is. I must have got confused between the names here and the names in riak_dt. Just a second while I update it in this branch.

**Comment by slfritchie** [comment]
Pulling PRs out of the freezer ... Sam, any chance that you could bribe a reviewer?  Was @coderoshi doing a full review or just a drive-by comment?

**Comment by russelldb** [comment]
@seancribbs is working on this in a different approach. Can wait for @lenary but I vote to close this.

**Comment by seancribbs** [comment]
@russelldb @slfritchie Yes, closing. See basho/yokozuna#250.

**Pull request closed without merging**